tests/integration/query.ts

```diff
@@ -112,7 +112,7 @@ describe('ensure correct results from latest request', () => {
   });
 
   it('vertical search get correct results based on up-to-date response', async () => {
-    answers.addListener({
+    const removeListener = answers.addListener({
```

Comment:
would it be a little simpler to just make a new instance of `answers` in each test? otherwise I think it's possible for people to add additional tests and forget to cleanup the listener.

I would probably just pull all the initialization into a function and call it at the start of each test. you could also use the whole pattern of re-initng it in the beforeEach, though I personally find that way a little more confusing (and I guess it's less "functional" with regards to functional programming)

```ts
let answers;
beforeEach(() => {
  answers = blah
})
```

Reply:
updated to the first approach so there's a function that's called at the start of each test to make a new instance of `answers`